app/controllers/api/base_controller.rb

```diff
@@ -46,6 +47,13 @@ class BaseController < ActionController::API
 
     private
 
+    def redirect_on_compressed_path
+      return unless [params[:c_id], params[:s_id]].any? { |id| Api.compressed_id?(id) }
+      url = request.original_url.sub(params[:c_id], Api.uncompress_id(params[:c_id]).to_s)
+      url.sub!(params[:s_id], Api.uncompress_id(params[:s_id]).to_s) if params[:s_id]
```

Comment:
? What's `sub`/`sub!`?

Reply:
Same as `gsub`/`gsub` but without the g - will substitute the first found match, and quit: https://ruby-doc.org/core-2.4.2/String.html#method-i-sub

Reply:
...I think my tags are broken again. 🙄 Thanks